src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionTests.cs

```diff
@@ -9535,14 +9928,43 @@ public static void Main()
     {
         System.Exception x = null;
         Method(x);
+        Method(ref x);
+        Method(in x);
+        Method(out x);
     }
 }";
+            CreateCompilation(code, parseOptions: TestOptions.Regular11).VerifyDiagnostics(
+                // (11,16): error CS1503: Argument 1: cannot convert from 'System.Exception' to 'in int'
+                //         Method(x);
+                Diagnostic(ErrorCode.ERR_BadArgType, "x").WithArguments("1", "System.Exception", "in int").WithLocation(11, 16),
+                // (12,20): error CS1615: Argument 1 may not be passed with the 'ref' keyword
```

Comment:
>// (12,20): error CS1615: Argument 1 may not be passed with the 'ref' keyword [](http://example.com/codeflow?start=16&length=77)

We might want to use a dedicated error message for this case, suggesting to upgrade language version to `C#12`. See ```ERR_NoImplicitConvTargetTypedConditional```, for example.